### ISSUE TITLE
TiDB Operator panic becuase of status.backoffRetryStatus

### ISSUE THREAD

**Issue by MarsBighead** [pingcap/tidb-operator#5584]
## Bug Report

**What version of Kubernetes are you using?**

```shell
kubectl version
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.2
```

**What version of TiDB Operator are you using?**

```
pingcap/tidb-backup-manager:v1.5.1
```
and local develop tidb-operator.

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

Pods run on Macbook air, and backup to minio(s3 compitable).

**What's the status of the TiDB cluster pods?**

```
kubectl -n tidb-admin get pod                                   
NAME                                                     READY   STATUS      RESTARTS         AGE
alpha-discovery-6cd7488955-jhgpb                         1/1     Running     1 (10d ago)      14d
alpha-pd-0                                               1/1     Running     1 (10d ago)      14d
alpha-tidb-0                                             2/2     Running     2 (8d ago)       14d
alpha-tikv-0                                             1/1     Running     1 (8d ago)       14d
```

**What did you do?**

Launch TiDB Operator on local Kubernates,  and launch local dev `controller-manager` connect to the K8S at same time. Actually, it will have 2 operator application to control resource by CR.

Then I submit a backup CR `alpha-backup-to-s3` to launch backup job. 
```yaml
apiVersion: pingcap.com/v1alpha1
kind: Backup
metadata:
  name: alpha-backup-to-s3
  namespace: tidb-admin
  labels:
    user: paul
    state: qa
spec:
  ## Describes the compute resource requirements and limits of Backup.
  ## Ref: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
  resources:
    requests:
      cpu: 500m
      memory: 512Mi
    limits:
      cpu: 500m
      memory: 512Mi

  ## List of environment variables to set in the container, like v1.Container.Env.
  ## Note that the following builtin env vars will be overwritten by values set here.
  ## - S3_PROVIDER
  ## - S3_ENDPOINT
  ## - AWS_REGION
  ## - AWS_ACL
  ## - AWS_STORAGE_CLASS
  ## - AWS_DEFAULT_REGION
  ## - AWS_ACCESS_KEY_ID
  ## - AWS_SECRET_ACCESS_KEY
  ## - GCS_PROJECT_ID
  ## - GCS_OBJECT_ACL
 ## From is the TidbCluster to be backed up.
  ## It takes high precedence than spec in BR. If `from` not set, cluster in BR will be backed up.
  #from:
    ## Host is the address of the TidbCluster to be backed up, which is the service name of the TidbCluster, such as `basic-tidb`.
    #host: alpha-tidb
    ## Port is the port of the TidbCluster to be backed up.
    #port: 4000
    ## User is the accessing user of the TidbCluster to be backed up.
    #user: root
    ## SecretName is the secret that contains the password of the accessing user of the TidbCluster to be backed up.
    # secretName: sh.helm.release.v1.tidb-operator.v1 
    ## TLSClientSecretName is the name of secret which stores tidb server client certificate.
    ## Defaults to nil.
    # tlsClientSecretName: ""
  backupType: full
  backupMode: snapshot
  ## TikvGCLifeTime specifies the safe gc life time for Backup.
  ## The time limit during which data is retained for each GC, in the format of Go Duration.
  ## When a GC happens, the current time minus this value is the safe point.
  ## Defaults to 72h.
  tikvGCLifeTime: 72h

  s3:
    provider: aws
    secretName: minio-secret
    region: shanghai
    bucket: tidbuss
    prefix: tidb/s3/alpha/concurrency/2
    endpoint: http://192.168.5.112:9000

  ## StorageSize is the PV size specified for the backup operation.
  ## This value must be greater than the size of the TidbCluster to be backed up.
  ## Defaults to 100Gi.
  storageSize: "10Gi"


  ## BR configuration.
  ## Ref: https://docs.pingcap.com/tidb/stable/backup-and-restore-tool
  br:
    ## Cluster specifies name of TidbCluster to be backed up.
    cluster: "alpha"
    ## Namespace specifies namespace of TidbCluster to be backed up.
    clusterNamespace: "tidb-admin"
    ## LogLevel is the log level. Defaults to `info`.
    # logLevel: "info"
    ## StatusAddr is the HTTP listening address for the status report service. Defaults to empty.
    # statusAddr: ""
    ## Concurrency is the size of thread pool on each node that execute the backup task.
    ## Defaults to 4.
    concurrency: 4
    ## RateLimit is the rate limit of the backup task, MB/s per node.
    ## If set to 4, the speed limit is 4 MB/s.The speed limit is not set by default.
    # rateLimit: 0
    ## TimeAgo presents back up the data before `timeAgo`, e.g. 1m, 1h. Defaults to empty.
    # timeAgo: 1m
    ## Checksum specifies whether to verify the files after the backup is completed.
    ## Defaults to `true``.
    # checksum: true
    ## CheckRequirements specifies whether to check requirements before backup
    # checkRequirements: true
    ## SendCredToTikv specifies whether the BR process passes its AWS or GCP privileges to the TiKV process.
    ## Defaults to `true``.
    sendCredToTikv: true
    ## OnLine specifies whether online during restore. Defaults to false.
    # onLine: false
    ## Options specifies the extra arguments that BR supports. These options has highest priority.
    # options: []

  

  ## ToolImage specifies the tool image used in `Backup`, which supports BR and Dumpling images.
  ## For examples `spec.toolImage: pingcap/br:v5.2.0` or `spec.toolImage: pingcap/dumpling:v5.2.0`
  ## For BR image, if it does not contain tag, Pod will use image 'ToolImage:${TiKV_Version}'.
  toolImage: pingcap/br:v6.5.5

  ## ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images.
  ## If private registry is used, imagePullSecrets may be set.
  ## You can also set this in service account.
  ## Ref: https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
  # imagePullSecrets:
  # - name: secretName

  ## TableFilter specifies tables that match the table filter rules for BR or Dumpling.
  ## Ref: https://docs.pingcap.com/tidb/stable/table-filter
  ## Defaults to empty.
  # tableFilter: []

  ## Affinity for Backup pod scheduling
  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
  # affinity: {}

  ## UseKMS to decrypt the secrets. Defaults to false.
  useKMS: false

  ## ServiceAccount Specify service account of Backup.
  serviceAccount: "tidb-backup-manager"

  ## CleanPolicy specifies whether to clean backup data when the Backup CR is deleted, if not set, the backup data will be retained.
  ## `Retain` represents that the backup data will be retained when the Backup CR is deleted.
  ## `OnFailure` represents that the backup data will be cleaned only for the failed backups when the Backup CR is deleted.
  ## `Delete` represents that the backup data will be cleaned when the Backup CR is deleted.
  cleanPolicy: Retain

```

**What did you expect to see?**

Both TiDB Operator and local dev `controller-manager`  running well without panic.

**What did you see instead?**

```
I0320 16:48:14.517923     299 backup_status_updater.go:129] Backup: [tidb-admin/alpha-backup-to-s3] updated successfully
I0320 16:48:14.517959     299 backup_controller.go:137] Finished syncing Backup "tidb-admin/alpha-backup-to-s3" (57.97325ms)
E0320 16:48:14.518032     299 backup_controller.go:118] Backup: tidb-admin/alpha-backup-to-s3, sync failed, err: create backup tidb-admin/alpha-backup-to-s3 job backup-alpha-backup-to-s3 failed, err: jobs.batch "backup-alpha-backup-to-s3" already exists, requeuing
I0320 16:48:14.522307     299 backup_controller.go:205] backup tidb-admin/alpha-backup-to-s3 is already Scheduled, Running, Preparing or Failed, skipping.
E0320 16:48:14.522460     299 runtime.go:78] Observed a panic: runtime.boundsError{x:-1, y:0, signed:true, code:0x0} (runtime error: index out of range [-1])
goroutine 239 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x105f15fe0, 0x14001354fa8})
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/runtime/runtime.go:74 +0xdc
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0})
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/runtime/runtime.go:48 +0xb0
panic({0x105f15fe0, 0x14001354fa8})
	/usr/local/go/src/runtime/panic.go:890 +0x230
github.com/pingcap/tidb-operator/pkg/controller/backup.(*Controller).detectBackupJobOrPodFailure(0x14000642c00, 0x14000b82400)
	/Users/hbu/operator/tidb-operator/pkg/controller/backup/backup_controller.go:293 +0x6e0
github.com/pingcap/tidb-operator/pkg/controller/backup.(*Controller).updateBackup(0x14000642c00, {0x10609e1e0, 0x14000b82400})
	/Users/hbu/operator/tidb-operator/pkg/controller/backup/backup_controller.go:214 +0xf38
github.com/pingcap/tidb-operator/pkg/controller/backup.NewController.func1({0x10609e1e0, 0x14001340400}, {0x10609e1e0, 0x14000b82400})
	/Users/hbu/operator/tidb-operator/pkg/controller/backup/backup_controller.go:63 +0x3c
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnUpdate({0x14000547190, 0x140005471a0, 0x140005471b0}, {0x10609e1e0, 0x14001340400}, {0x10609e1e0, 0x14000b82400})
	/Users/hbu/go/pkg/mod/k8s.io/client-go@v0.20.15/tools/cache/controller.go:238 +0x5c
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/Users/hbu/go/pkg/mod/k8s.io/client-go@v0.20.15/tools/cache/shared_informer.go:775 +0x1a4
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x14000df8f00)
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:155 +0x48
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x14000f99f00, {0x1060fa040, 0x14000b03350}, 0x1, 0x14000c45bc0)
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:156 +0xa0
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x14000df8f00, 0x3b9aca00, 0x0, 0x1, 0x14000c45bc0)
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:133 +0x78
k8s.io/apimachinery/pkg/util/wait.Until(0x14000df8f00, 0x3b9aca00, 0x14000c45bc0)
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:90 +0x34
k8s.io/client-go/tools/cache.(*processorListener).run(0x14000d18980)
	/Users/hbu/go/pkg/mod/k8s.io/client-go@v0.20.15/tools/cache/shared_informer.go:771 +0x78
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:73 +0x84
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/Users/hbu/go/pkg/mod/k8s.io/apimachinery@v0.20.15/pkg/util/wait/wait.go:71 +0xb8
```

**Comment by csuzhangxc** [comment]
@MarsBighead Thanks for your investigation, I think your suggestion is also perfect. Cloud you open a PR to fix it?

**Comment by csuzhangxc** [comment]
closed by #5597
